server/base/src/main/java/org/apache/accumulo/server/ServerContext.java

```diff
@@ -113,6 +114,7 @@ private ServerContext(ServerInfo info) {
     serverDirs = info.getServerDirs();
 
     propStore = memoize(() -> ZooPropStore.initialize(getInstanceID(), getZooReaderWriter()));
+    zkUserPath = memoize(() -> Constants.ZROOT + "/" + getInstanceID() + Constants.ZUSERS);
```

Comment:
I don't think you need to use memoize here. That's for lazy loading instances with a supplier to load when needed but the zkUserPath is just a final String value so it should be fine to just use directly, ie `zkUserPath = Constants.ZROOT + "/" + getInstanceID() + Constants.ZUSERS;`

Reply:
Oops just missed the early comments that suggested we SHOULD be using that so I guess ignore my comment :)

Reply:
I recommended memoize because `getInstanceID()` may not be ready right away. We have had some bootstrapping issues with this, especially around the Initialize code. I could have been overly cautious in my recommendation, though. It might not actually be needed here. If it's not, I'm fine with dropping that bit and just making it a final String.